### PR TITLE
Fix typos in docs/

### PR DESCRIPTION
**Pull request by nikosavola**
As suggested in https://github.com/scikit-rf/scikit-rf/pull/484#pullrequestreview-692073778, this PR fixes typos in the documentation notebooks (and one `.rst`) using the same method as https://github.com/scikit-rf/scikit-rf/pull/484, i.e., scoping typos automatically with [scspell](https://pypi.org/project/scspell/).

### CODE REVIEW

File: doc/source/examples/networktheory/Working with Complex Characteristic Impedances.ipynb [hunk 1]
```diff
@@ -71,7 +71,7 @@
     "### Power-waves\n",
     "The scattering matrix as defined above has the fundamental problem that a zero-reflection coefficient does not guarantee maximal power transfer from the generator to the load [5-6]. To remedy this situation, the concept of _power-waves_ has been introduced [4]. It is a mathematical tool which is compatible with the conjugate matching condition. It also gives intuitive total power expressions, as the sum of forward and backward waves, also for complex characteristic impedances. \n",
     "\n",
-    "The power-waves $a$ and $b$ are defined from frequency dependant (complex) port impedance $Z_0$ [4]:\n",
+    "The power-waves $a$ and $b$ are defined from frequency dependent (complex) port impedance $Z_0$ [4]:\n",
```
Comment (Vinc0110) [inline doc/source/examples/networktheory/Working with Complex Characteristic Impedances.ipynb:74]: frequency-dependent
(I know this is very picky, sorry)

**Review by Vinc0110 — APPROVED** [review]
That's one very useful piece of software. Thanks for the fixes!

It's funny how almost every notebook contained at least one misspelled "necessary" (including my own :dagger:)

**Comment by jhillairet** [comment]
Great! OK for me to be merged.

**Comment by nikosavola** [comment]
Implemented the aforementioned hyphen change by @Vinc0110. Can be merged now.

**Pull request merged**